doc/source/dev/contributor/rendering_documentation.rst

```diff
@@ -217,6 +217,75 @@ Some examples:
         Some text that follows the list.
 
 
+Self-contained examples
+~~~~~~~~~~~~~~~~~~~~~~~
+All examples must be self-contained. It means all imports must be explicit,
+the data used must be defined in the examples, etc.
+
+    Yes::
+
+        >>> import numpy as np
+        >>> rng = np.random.default_rng()
+
+    No::
+
+        >>> rng = np.random.default_rng()
+
+The variable scope is defined
+per docstrings. i.e. for a given function's docstrings. It means variables
+cannot be shared between functions from the same file. What is possible,
+and recommended, is to separate code within a single example block with
+explanations. Note that blank lines surrounding all block codes are required.
+
+    Yes::
+
+        Some initial text
+
+        >>> import numpy as np
+        >>> rng = np.random.default_rng()
+
+        This is some explanation
+
+        >>> rng.random(10)
+
+
+Examples and randomness
+~~~~~~~~~~~~~~~~~~~~~~~
+Examples are executed in the CI and the output is compared against the provided
+reference. The main goal is to ensure that the API is correctly documented and
+that we are not using outdated syntax. Doctests are not meant to be used as
+unittests.
```

Comment:
Doctests can't directly verify that the "API is correctly documented", though. It can only verify that the example itself is working as intended. How about:
```suggestion
In the continuous integration (CI) suite, examples are executed and the output
is compared against the provided reference. The main goal is to ensure that
the *example* is correct; a failure warns us that the example may need to be
adjusted (e.g. because the API has changed since it was written).
Doctests are not meant to be used as unit tests of underlying implementation.
```